src/pvrclient-nextpvr.cpp

```diff
@@ -1317,10 +1326,23 @@ bool cPVRClientNextPVR::UpdatePvrRecording(TiXmlElement* pRecordingNode, PVR_REC
   if (tag->channelType != PVR_RECORDING_CHANNEL_TYPE_RADIO)
   {
     char artworkPath[512];
-    snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=recording.artwork&sid=%s&recording_id=%s", g_szHostname.c_str(), g_iPort, m_sid, tag->strRecordingId);
-    PVR_STRCPY(tag->strThumbnailPath, artworkPath);
-    snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=recording.fanart&sid=%s&recording_id=%s", g_szHostname.c_str(), g_iPort, m_sid, tag->strRecordingId);
-    PVR_STRCPY(tag->strFanartPath, artworkPath);
+    if (m_backendVersion < 50000)
+    {
+      snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=recording.artwork&sid=%s&recording_id=%s", g_szHostname.c_str(), g_iPort, m_sid, tag->strRecordingId);
+      PVR_STRCPY(tag->strThumbnailPath, artworkPath);
+      snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=recording.fanart&sid=%s&recording_id=%s", g_szHostname.c_str(), g_iPort, m_sid, tag->strRecordingId);
+      PVR_STRCPY(tag->strFanartPath, artworkPath);
+    }
+    else
+    {
+      if (g_sendSidWithMetadata)
+        snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=channel.show.artwork&sid=%s&name=%s", g_szHostname.c_str(), g_iPort, m_sid, UriEncode(tag->strTitle).c_str());
+      else
+        snprintf(artworkPath, sizeof(artworkPath), "http://%s:%d/service?method=channel.show.artwork&name=%s", g_szHostname.c_str(), g_iPort, UriEncode(tag->strTitle).c_str());
+      PVR_STRCPY(tag->strFanartPath, artworkPath);
+      strcat(artworkPath, "&prefer=poster");
+      PVR_STRCPY(tag->strThumbnailPath, artworkPath);
+    }
```

Comment:
Maybe a future item to remove the horrible PVR_STRCPY macro usage.

Reply:
That is definitely in may plans as I move from the pvr-vuplus style coding I started with Settings  I also will run clang-format on all the new modules to deal with the style issues.